### PR TITLE
Disable autocorrect and autocaptialisation when entering username for SSO registration.

### PR DESCRIPTION
**Pull request by pixlwave**
When registering a new account via SSO on iOS, the text field becomes pretty annoying as it autocapitalises and autocorrects your input. This PR fixes that (although I have only tested the raw HTML file on the simulator, I'm not sure how to get the complete setup available for testing in the flow).

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: synapse/res/templates/sso_auth_account_details.html [hunk 1]
```diff
@@ -138,7 +138,7 @@ <h1>Create your account</h1>
         <div class="username_input" id="username_input">
           <label for="field-username">Username (required)</label>
           <div class="prefix">@</div>
-          <input type="text" name="username" id="field-username" value="{{ user_attributes.localpart }}" autofocus>
+          <input type="text" name="username" id="field-username" value="{{ user_attributes.localpart }}" autofocus autocorrect="off" autocapitalize="none">
```
Comment (clokep) [inline synapse/res/templates/sso_auth_account_details.html:141]: I see `autocorrect` at https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#non-standard_attributes, but not `autocapitalize`. Do you have a reference for that?

Comment (pixlwave) [reply synapse/res/templates/sso_auth_account_details.html:141]: Yep, theres one over at Apple here: https://developer.apple.com/library/archive/documentation/AppleApplications/Reference/SafariHTMLRef/Articles/Attributes.html#//apple_ref/doc/uid/TP40008058-autocapitalize

**Comment by pixlwave** [comment]
Looking at the failed complement tests, I'm presuming those wouldn't have been affected by this PR?

**Comment by clokep** [comment]
> Looking at the failed complement tests, I'm presuming those wouldn't have been affected by this PR?

It won't be, some of those tests are flaky right now -- #13161.

File: changelog.d/13350.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+Disabled autocorrection and autocapitalisation on the username text field shown during registration when using SSO.
```
Comment (erikjohnston) [inline changelog.d/13350.bugfix:1]:
```suggestion
Disable autocorrection and autocapitalisation on the username text field shown during registration when using SSO.
```

**Review by erikjohnston — APPROVED** [review]

**Pull request merged**